src/main.cpp

```diff
@@ -3881,10 +3886,11 @@ TbBool tag_cursor_blocks_place_room(PlayerNumber plyr_idx, MapSubtlCoord stl_x,
     {
         map_volume_box.visible = 1;
         map_volume_box.color = allowed;
-        map_volume_box.beg_x = subtile_coord((render_roomspace.left * 3), 0);
-        map_volume_box.beg_y = subtile_coord((render_roomspace.top * 3), 0);
-        map_volume_box.end_x = subtile_coord((3*a4) + (render_roomspace.right * 3), 0);
-        map_volume_box.end_y = subtile_coord(((3*a4) + render_roomspace.bottom * 3), 0);
+        // Draw "outer"/"square" boundbox ("accurate" boundbox is stored in the roomspace)
+        map_volume_box.beg_x = subtile_coord(((render_roomspace.centreX - calc_distance_from_roomspace_centre(render_roomspace.width,0)) * 3), 0);
+        map_volume_box.beg_y = subtile_coord(((render_roomspace.centreY - calc_distance_from_roomspace_centre(render_roomspace.height,0)) * 3), 0);
+        map_volume_box.end_x = subtile_coord((3*a4) + ((render_roomspace.centreX + calc_distance_from_roomspace_centre(render_roomspace.width,(render_roomspace.width % 2 == 0))) * 3), 0);
+        map_volume_box.end_y = subtile_coord(((3*a4) + (render_roomspace.centreY + calc_distance_from_roomspace_centre(render_roomspace.height,(render_roomspace.height % 2 == 0))) * 3), 0);
```

Comment:
What is 3? It looks like it is STL_PER_SLAB.
What is a4? it is a kind of slab coords? could you find what it exactly is and rename.

Reply:
You are right about 3, it is STL_PER_SLAB.

I am not sure about a4, but I'll try and find out.

Reply:
`a4` is `player->field_4A4`, so I am working on identifying `player->field_4A4` by its use. My current label is "valid_slab_under_cursor", but it could be "show_cursor" or something else, it certainly seems to be a boolean by it's use in KFX. 

`player->field_4A4`'s use in main.cpp seems redundant (`player->field_4A4` seems to always be 1 before it is passed as a parameter to the functions in main.cpp), and could be a decompilation artefact - but I am checking over this too.

Reply:
There are also some other unnamed fields (e.g. field_4AB, field_4AD) that I will try and name, as their scope seems small, and adjacent to player->field_4A4

Reply:
`player->field_4A4` is `player->full_slab_cursor` (please suggest a better name than "full_slab_cursor" based on the following description)
IF `player->full_slab_cursor` is equal to 0, then a single subtile bounding box is drawn
IF `player->full_slab_cursor` is equal to 1, then a single slab bounding box is drawn

Using this DLL function, with these amended parameters will give a subtile digging cursor: ```_DK_tag_cursor_blocks_dig(plyr_idx, stl_x, stl_y, 0);```
![image](https://user-images.githubusercontent.com/1984342/113304333-d6038900-92f9-11eb-902a-50782496c3d5.png)
If someone could test that 1 line change from an earlier version of KFX (before one-click or single subtile selling), that would be great. Search for "_DK_tag_cursor_blocks_dig(plyr_idx" in main.cpp and replace the line in tag_cursor_blocks_dig() with the one above the image.
You should get the same results, as per the image.

Reply:
Should be resolved as of 215f650.

@SimLV and @Loobinex please check over the renamed properties and their use.

Reply:
I checked and have no issues with the renamed properties. No need to put the old name in a comment behind it though if it was some randomly generated fieldname.